### ISSUE TITLE
React list form: Date picker no longer works

### ISSUE THREAD

**Issue by JenniferUXWeb** [pnp/sp-dev-fx-webparts#1451]
Sample (which sample are you having trouble with)
**React list form**

Authors
**Dany Wyss @harshagracy**

Expected or Desired Behavior
When a form contains a date field, user should be able to choose a date using the date picker or manually type a valid date into the input field.

Observed Behavior
Date field displays 'invalid date' placeholder text and does not allow user to add a date using the date picker or type a date manually. If a date is selected the entire form disappears and an error message is displayed. 

![image](https://user-images.githubusercontent.com/68618534/90568713-5b3f9080-e1a4-11ea-9819-135fc9a6542d.png)

Error message after picking a date:
**Error: Minified React error #185; visit https://reactjs.org/docs/error-decoder.html?invariant=185 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.**

![image](https://user-images.githubusercontent.com/68618534/90568831-95a92d80-e1a4-11ea-868b-26b18973cd66.png)

Steps to Reproduce

1. Deploy React list web part to App catalog
2. Add web part to page
3. Configure settings and add a list with a date field

 Environment Details (*Development & Target environment*)

- **OS**: Windows 10
- **Target Environment**: SharePoint Online
- **Framework**:  Node.js v10
- **Browser(s)**: Chrome v79 | Safari v22 | FireFox v31 | IE v11 | Edge | Chromium Edge v79
- **Tooling**: SPFx v1.10.0 
- **Additional details**: *The more context you can provide, the easier it is (and therefore quicker) to help.*

**Comment by PathToSharePoint** [comment]
I just did a test on the workbench with a date field and it looks fine on my end (screenshot below).

I see another error message on your screenshot unsupported field type "User". Could that be the actual issue?
![image](https://user-images.githubusercontent.com/40323257/91108432-ee594a00-e62c-11ea-815c-95f24ff36a6f.png)

**Comment by JenniferUXWeb** [comment]
> I just did a test on the workbench with a date field and it looks fine on my end (screenshot below).
> 
> I see another error message on your screenshot unsupported field type "User". Could that be the actual issue?
> 
> ![image](https://user-images.githubusercontent.com/40323257/91108432-ee594a00-e62c-11ea-815c-95f24ff36a6f.png)

Thank you so much for taking the time to look into this @PathToSharePoint . I checked again and it looks like it's definitely the date field causing the issue for me.

Could it be a browser issue I wonder? I'm using a Chrome version that is slightly out of date.

I've checked the error message on the React website and here are the full details below:

**Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.**

![image](https://user-images.githubusercontent.com/68618534/91659792-0bc56280-eaca-11ea-997a-978ad2790fdb.png)

**Comment by PathToSharePoint** [comment]
Thanks @JenniferUXWeb , that's very useful information.

I did some research and found out that the issue might be linked to the use of the "moment" library in this sample. I checked out the Fluent UI site and it doesn't seem to use moment, so I believe this was introduced by the solution author @Harshagracy .

Let's see what the author has to say about that. If we don't hear back from the author I can take a closer look. My issue right now is that I can't reproduce the bug.

Good point about the browser too, it's true that Microsoft only commits to supporting the latest version of Chrome.
